daemons/pacemakerd/pcmkd_messages.c

```diff
@@ -123,23 +168,8 @@ pcmk_ipc_dispatch(qb_ipcs_connection_t * qbc, void *data, size_t size)
 
     task = crm_element_value(msg, F_CRM_TASK);
     if (pcmk__str_eq(task, CRM_OP_QUIT, pcmk__str_none)) {
-        /* Only allow privileged users (i.e. root or hacluster) to shut down
-         * Pacemaker from the command line (or direct IPC), so that other users
-         * are forced to go through the CIB and have ACLs applied.
-         */
-        bool allowed = pcmk_is_set(c->flags, pcmk__client_privileged);
-
-        if (allowed) {
-            crm_notice("Shutting down in response to IPC request %s from %s",
-                       crm_element_value(msg, F_CRM_REFERENCE),
-                       crm_element_value(msg, F_CRM_ORIGIN));
-            pcmk__ipc_send_ack(c, id, flags, "ack", CRM_EX_OK);
-            pcmk_shutdown(15);
-        } else {
-            crm_warn("Ignoring shutdown request from unprivileged client %s",
-                     pcmk__client_name(c));
-            pcmk__ipc_send_ack(c, id, flags, "ack", CRM_EX_INSUFFICIENT_PRIV);
-        }
+        pcmk__ipc_send_ack(c, id, flags, "ack", CRM_EX_INDETERMINATE);
```

Comment:
I don't have an explanation for why the ping ACK wasn't tripping up dispatch, either, unless it's something I somehow added with these patches.  The message I am seeing is "Bad reply from pacemakerd: Protocol violated", which is coming out of the new pacemakerd_event_cb function I added.  I modeled that after the same function in crm_mon.c, but perhaps it's incorrect.

Reply:
Without the ACK message, pcmk__send_ipc_request returns ENODATA, because crm_ipc_send returns 0 because it is timing out.  The shutdown still happens, though.  With the ACK message, pcmk__send_ipc_message completes almost instantaneously and the logs look like they should.  So, the ACK message seems pretty important.